pkg/operator/configmetrics/configmetrics.go

```diff
@@ -49,11 +49,22 @@ func (m *configMetrics) Create(version *semver.Version) bool {
 
 // Describe reports the metadata for metrics to the prometheus collector.
 func (m *configMetrics) Describe(ch chan<- *prometheus.Desc) {
-	ch <- m.cloudProvider.WithLabelValues("", "").Desc()
+	ch <- m.cloudProvider.WithLabelValues("", "", "").Desc()
 	ch <- m.featureSet.WithLabelValues("").Desc()
 	ch <- m.proxyEnablement.WithLabelValues("").Desc()
 }
 
+// Returns external platform name if present, empty string otherwise
+func getExternalPlatformName(infra *configv1.Infrastructure) string {
+	if infra == nil {
+		return ""
+	}
+	if infra.Spec.PlatformSpec.External == nil {
+		return ""
+	}
```

Comment:
```suggestion
	if infra == nil || infra.Spec.PlatformSpec.External == nil {
		return ""
	}
```

Reply:
good point but this is a backport, should we make it different than what's in 4.15 and 4.16?

Reply:
It makes sense.